### PR TITLE
Patch to let players drop items

### PR DESCRIPTION
**Pull request by VictorD**
This lets players drop items by pressing Q or throwing them outside their inventory.

Block collision is kind of temporary (and only in Y-axis) since I don't know how to use the real stuff and gravity is just -0.04f applied each tick, since GravityAction is meant for mobs(?) :)

Signed-off-by: Victor Danell victor.danell@gmail.com

### CODE REVIEW

File: src/main/java/org/spout/vanilla/protocol/handler/DiggingMessageHandler.java [hunk 1]
```diff
@@ -97,6 +99,10 @@ public void handleServer(Session session, Player player, DiggingMessage message)
 		} else if (message.getState() == DiggingMessage.STATE_DONE_DIGGING) {
 			//TODO: Timing checks!
 			blockBroken = true;
+		} else if (message.getState() == DiggingMessage.STATE_DROP_ITEM && isVanilla) {
+			VanillaPlayer vplayer = (VanillaPlayer) player.getEntity().getController();
+			if (vplayer != null)
```
Comment (Top-Cat) [inline src/main/java/org/spout/vanilla/protocol/handler/DiggingMessageHandler.java:104]: OTB add the brackets

Few other places too

File: src/main/java/org/spout/vanilla/controller/living/player/VanillaPlayer.java [hunk 2]
```diff
@@ -492,4 +496,83 @@ public ItemStack getItemOnCursor() {
...
+
+		ItemStack dropStack = stack.clone();
+		dropStack.setAmount(amount);
+		Item newItem = new Item(dropStack, velocity);
+		Vector3 position = e.getPosition().add(itemPositionOffset);
+		e.getWorld().createAndSpawnEntity(new Point(position, getParent().getWorld()), newItem);
+		return newItem;
+	}
+
+	private static ItemStack subtractFromStack(ItemStack in, int count) {
```
Comment (ZNickq) [inline src/main/java/org/spout/vanilla/controller/living/player/VanillaPlayer.java:567]: This shouldn't be in VanillaPlayer,more like SpoutAPI...

File: src/main/java/org/spout/vanilla/controller/living/player/VanillaPlayer.java [hunk 3]
```diff
@@ -218,6 +219,9 @@ public PlayerInventory createInventory(int size) {
 		}
 
 		inventory.setCurrentSlot(0);
+		if (getActiveInventory() == null)
```
Comment (ZNickq) [inline src/main/java/org/spout/vanilla/controller/living/player/VanillaPlayer.java:222]: Pretty sure createInventory sets the inventory immediately afterwards...was there an issue with this?

Comment (VictorD) [inline src/main/java/org/spout/vanilla/controller/living/player/VanillaPlayer.java:222]: Ah I see. Without that line I couldn't press Q after joining and grabbing an item but had to open the inventory, lift it up and put it down again so I figured it wasn't set

File: src/main/java/org/spout/vanilla/protocol/handler/DiggingMessageHandler.java (hunk 1 above)
Comment (VictorD) [inline src/main/java/org/spout/vanilla/protocol/handler/DiggingMessageHandler.java:104]: OTB? I assume it means Omitting The Brackets or something , I've added them locally now

Comment (Top-Cat) [inline src/main/java/org/spout/vanilla/protocol/handler/DiggingMessageHandler.java:104]: No, one true brace is the formatting style

**Pull request closed without merging**